app/exercises/base.py

```diff
@@ -53,4 +53,7 @@ def clear_text(self):
 
     def clear_graph(self):
         self.graph = None
-        self.canvas.delete('all')
\ No newline at end of file
+        self.canvas.delete('all')
+
+    def add_text_to_result(self, text):
```

Comment:
There are no consequences here clear_graph, draw_graph, print_graph, add_text_to_result. Names are more than misleading. Maybe change to something like that:
- draw_graph (no changes)
- clear_graph (no changes)
- print_text
- clear_text
- append_text
Or give other suggestions

Reply:
print_graph is good (it prints graph, not text from parameter)

Reply:
append_text as method in BaseTab class says nothing

Reply:
But it doesnt print only graph.  It print also paths and results. U can change "text" to something better

Reply:
No, it prints only graph (list)